projects/OG-Analytics/src/main/java/com/opengamma/analytics/math/interpolation/GridInterpolator2D.java

```diff
@@ -67,6 +70,39 @@ public GridInterpolator2D(final Interpolator1D xInterpolator, final Interpolator
     ArgumentChecker.notNull(data, "data");
     return testData(data);
   }
+  
+  public Map<Double, Interpolator1DDataBundle> getDataBundleFromUnsorted(final double[] xInput, final double[] yInput, final double[] zInput, boolean mustCopy) {
+    ArgumentChecker.notNull(xInput, "xValues");
+    ArgumentChecker.notNull(yInput, "yValues");
+    ArgumentChecker.notNull(zInput, "zValues");
+    
+    double[] xValues = mustCopy ? xInput.clone() : xInput;
+    double[] yValues = mustCopy ? yInput.clone() : yInput;
+    double[] zValues = mustCopy ? zInput.clone() : zInput;
+    
+    Map<Double, Interpolator1DDataBundle> result = new TreeMap<Double, Interpolator1DDataBundle>();
+    
+    ParallelArrayBinarySort.parallelBinarySort(xValues, yValues, zValues);
+    Double currX = xValues[0];
+    long currXBits = Double.doubleToRawLongBits(currX);
+    int currStartRange = 0;
+    for (int i = 0; i <= xValues.length; i++) {
+      if ((i == xValues.length) || (currXBits != Double.doubleToRawLongBits(xValues[i]))) {
+        //ParallelArrayBinarySort.parallelBinarySort(yValues, zValues, currStartRange, i - 1);
```

Comment:
It's WIP in that instance. I'll be able to resolve that and a few other things when the new interpolated surface is committed.